feat/completion-pum

mosheavni opened the pull request:
fixes #357 

WORK IN PROGRESS
<img width="590" alt="image" src="https://github.com/user-attachments/assets/fca42c47-255a-45ce-80e5-22d9dc58aca5">

mosheavni commented:
currently:
![2024-09-25_14-34-49 (1)](https://github.com/user-attachments/assets/50eba495-3174-42d7-8252-8164526f3280)

issues:

- [ ] long results are currently not shown (more than 50)
- [ ] pum stays visible for a few seconds after the selection is chosen
- [ ] vim.on_key is on every buffer. how can I limit to only the prompt buffer (alias/ns/ctx)?

@Ramilito  need ur help here 🙏🏼
you'd probably also want to move stuff around

Ramilito commented:
The gifs are looking soooo goood!
I'll try to get some time later tonight and look if I can help

mosheavni commented:
> The gifs are looking soooo goood!
> I'll try to get some time later tonight and look if I can help

Please work your magic here because I am really stuck and you seem to get the hang of all those UI components

Ramilito commented:
>  long results are currently not shown (more than 50)
 pum stays visible for a few seconds after the selection is chosen
 vim.on_key is on every buffer. how can I limit to only the prompt buffer (alias/ns/ctx)?

- Long results, I removed the limit you added and added a windows height restriction instead, so we would scroll for the rest.
- Not seeing the delay, is it still there?
- vim.on key wasn't called because the buffer wasn't really removed (just hidden), I set it to wipe on close and now it's getting called

Can you verify that I understood this correctly?

mosheavni commented:
yes, it looks great :)
I changed row to 1 to move the pum a line lower so it won't hide the text, and changed some stuff around the highlights. 
ok to merge? :)

Ramilito commented on lua/kubectl/actions/highlight.lua:
```diff
@@ -31,12 +31,13 @@ local highlights = {
 function M.setup()
   for group, attrs in pairs(highlights) do
-    local success, hl = pcall(api.nvim_get_hl_by_name, group, true)
+    local success, hl = pcall(api.nvim_get_hl, 0, group)
```
nice!

Ramilito commented:
It looks sooo cool!
Love your work here!
